platform/viewer/src/connectedComponents/ViewerRetrieveStudyData.js

```diff
@@ -83,18 +85,56 @@ class ViewerRetrieveStudyData extends Component {
     });
   }
 
+  _handleSeriesLoadResult(error, studyMetadata, series) {
+    if (this.abortSeriesLoad) return;
+    const seriesLoadStats = this.seriesLoadStats[
+      studyMetadata.getStudyInstanceUID()
+    ];
+    if (!seriesLoadStats) return;
+    seriesLoadStats.count--;
+    if (error || !series) {
+      seriesLoadStats.errors++;
+      log.error(error || 'Bad Series');
+      return;
+    }
+    this._addSeriesToStudy(studyMetadata, series);
+  }
+
   _attemptToLoadRemainingSeries(studyMetadata) {
     const { seriesLoader } = studyMetadata.getData();
     if (!seriesLoader) {
       return;
     }
+    const seriesLoadStats = (this.seriesLoadStats[
+      studyMetadata.getStudyInstanceUID()
+    ] = {
+      errors: 0,
+      count: 0,
+    });
```

Comment:
Track each success/fail for each study's series metadata load?

Reply:
yes!